src/main/java/org/broadinstitute/dsm/shipping/UPSTracker.java

```diff
@@ -43,12 +42,26 @@ public UPSTrackingResponse lookupTrackingInfo(String trackingId) {
         headers.put("AccessLicenseNumber", accessKey);
         headers.put("Content-Type", "application/json");
         headers.put("Accept", "application/json");
-        try {
-            UPSTrackingResponse response = DDPRequestUtil.getResponseObjectWithCustomHeader(UPSTrackingResponse.class, sendRequest, "UPS Tracking Test " + inquiryNumber, headers);
-            return response;
+
+        RuntimeException error = null;
+        for (int i = 0; i < 5; i++) {
+            try {
+                UPSTrackingResponse response = DDPRequestUtil.getResponseObjectWithCustomHeader(UPSTrackingResponse.class, sendRequest, "UPS Tracking Test " + inquiryNumber, headers);
+                return response;
+            }
+            catch (Exception e) {
+               error = new RuntimeException("couldn't get response from ups tracking of package " + trackingId, e);
+               logger.warn("Retrying ups round " + i);
+               try {
+                   Thread.sleep(5 * 1000);
+               } catch (InterruptedException interrupted) {
+                   logger.warn("Error while waiting for UPS retry", interrupted);
+               }
+            }
         }
-        catch (IOException e) {
-            throw new RuntimeException("couldn't get response from ups tracking of package " + trackingId, e);
+        if (error != null) {
```

Comment:
Shouldn’t `error` get null every time we enter the `for` loop? So that we don't throw an error if we had one in the first try but was successful in the 2nd?

Reply:
it does. L46

Reply:
That's before the loop though

Reply:
The early return on L50 will ignore any exceptions from previous errors.  If we fail 5 times, L63 will throw the last error from the latest attempt.  Does that sound about right?

Reply:
yes

Reply:
Yes that's right. I missed the `return` part. Thanks for the explanation @zyme